lib/foreman_discovery/lldp_neighbors.rb

```diff
@@ -0,0 +1,65 @@
+module ForemanDiscovery
+  class LldpNeighbors
+    def set(iface, neighbor)
+      interfaces[iface] = neighbor
+    end
+
+    def get(iface)
+      interfaces[iface]
+    end
+
+    def list_by_pvid
+      list = {}
+      interfaces.each do |name, neighbor|
+        next unless neighbor.has_key? 'PVID'
+        vlan = neighbor['PVID']
+        list[vlan] = [] unless list.has_key? vlan
+        list[vlan].push name
+      end
+
+      list
+    end
+
+    def interfaces
+      @interfaces ||= {}
+    end
+
+    def get_neighbors_by_interface(ifname)
+      return nil unless interfaces.has_key? ifname
+
+      interface = get(ifname)
+      return nil unless interface.has_key? 'PVID'
+
+      list = []
+      interfaces.each do |name, neighbor|
+        next unless neighbor.has_key? 'PVID'
+        list.push name if neighbor['PVID'] == interface['PVID']
+      end
+
+      list.size > 1 ? list : nil
```

Comment:
Nit:
`list.empty? ? nil : list`

or

```ruby
return if list.empty?
list
```

Reply:
Not sure... I had the second one first, but I hate it to read code where it's not immediately obvious that nil/null must be expected. And what's ugly about this is that it's not `list.empty?` - only results with more than one entry are returned, if there is only one `peer` it should return `nil`.

I changed it to read:
```ruby
      return if list.size < 2
      list
```

This way it's more obvious that at least two are required for a valid candidate.

Cheers,
Thomas